stdlib/src/builtin/dtype.mojo

```diff
@@ -641,6 +641,24 @@ fn _integral_type_of[type: DType]() -> DType:
     return type.invalid
 
 
+fn _scientific_notation_digits[type: DType]() -> StringLiteral:
+    """Get the number of digits for the scientific notation representation of a float
+    """
+
+    @parameter
+    if type == DType.bfloat16 or type == DType.float16:
+        return "4"
+
+    @parameter
+    if type == DType.float32 or type == DType.tensor_float32:
+        return "8"
+
+    @parameter
+    if type == DType.float64:
+        return "16"
+    return "0"
```

Comment:
Is it possible to use `constrained` here?
```mojo
constrained[type.is_floating_point(), ...]
```

Reply:
There doesn't seem to be a way to prevent line 658 in `simd.mojo` from evaluating in so it'll produce compilation errors when calling `__repr__` with integral and other types?

Reply:
Yeah, you probably have to guard the callsite with a parameter if.

Reply:
Oh right silly me I had to add `type.is_floating_point()` to the condition but that works now!